File: Python/1863.py
# 1863. Sum of All Subset XOR Totals
# https://leetcode.com/problems/sum-of-all-subset-xor-totals/

from functools import reduce

class Solution:
  def subsetXORSum(self, nums: List[int]) -> int:
    return reduce(lambda x, y: x | y, nums) << (len(nums) - 1)

File: Rust/1863.rs
/**
 * 1863. Sum of All Subset XOR Totals
 * https://leetcode.com/problems/sum-of-all-subset-xor-totals/
 */

impl Solution {
    pub fn subset_xor_sum(nums: Vec<i32>) -> i32 {
        let shift = nums.len() - 1;

        return (
            nums
                .into_iter()
                .reduce(|a, b| a | b)
                .unwrap() << shift
        );
    }
}
